### PR TITLE
lldb: link swift's libswift to lldb

### PR DESCRIPTION
**Pull request by eeckstein**
This is needed for the swift expression parser once parts of the parser are implemented in swift.
For details on libswift see https://github.com/apple/swift/blob/main/libswift/README.md.

### CODE REVIEW

File: lldb/cmake/modules/AddLLDB.cmake [hunk 1]
```diff
@@ -151,6 +151,36 @@ function(add_lldb_library name)
   endif()
 endfunction(add_lldb_library)
 
+# BEGIN Swift Mods
+function(add_properties_for_libswift target)
+  if (LIBSWIFT_BUILD_MODE)
+    if (CMAKE_SYSTEM_NAME MATCHES "Darwin")
+      if(LIBSWIFT_BUILD_MODE MATCHES "HOSTTOOLS|.*HOSTLIBS")
+        target_link_directories(${target} PRIVATE
+            "${CMAKE_OSX_SYSROOT}/usr/lib/swift"
```
Comment (adrian-prantl) [inline lldb/cmake/modules/AddLLDB.cmake:160]: This line looks suspicious to me but I also don't fully understand what it does. Does this end up being in an rpath pointing into the build machine's SDK?

Comment (eeckstein) [reply lldb/cmake/modules/AddLLDB.cmake:160]: For macOS: only in "bootstrapping" mode, which is only used for testing. For other modes the rpath is pointing to `/usr/lib/swift`.

For Linux: yes, but I couldn't figure out how to fix that. But this is already the case without this change.

**Comment by eeckstein** [comment]
@swift-ci test

**Comment by eeckstein** [comment]
@swift-ci test

**Comment by eeckstein** [comment]
@swift-ci test

**Review by JDevlieghere — APPROVED** [review]
LGTM

**Comment by eeckstein** [comment]
@swift-ci test

**Pull request merged**